### PR TITLE
Allow filtering on full names in mention HUD

### PR DESCRIPTION
**Pull request by buoyad**
r? @keybase/react-hackers

### CODE REVIEW

File: shared/chat/hud/mention-hud-container.js [hunk 1]
```diff
@@ -16,9 +16,9 @@ type ConnectedMentionHudProps = {
 
 const mapStateToProps: MapStateToProps<*, *, *> = state => {
   const inbox = getSelectedInbox(state)
-  const participants = inbox ? inbox.get('participants').toArray() : []
+  const users = inbox ? inbox.get('fullNames').toArray() : []
```
Comment (chrisnojima) [inline shared/chat/hud/mention-hud-container.js]: lets not use array of arrays. either make it a {[key: name: string} or make an actual object

File: shared/chat/hud/index.js [hunk 2]
```diff
@@ -87,16 +87,20 @@ type MentionHudProps = {
...
-      fullName: '', // TODO
-      key: u,
+      username: u[0],
+      fullName: u[1],
+      key: u[0],
     }))
-    .filter(u => u.username.indexOf(filter) >= 0)
+    .filter(
+      u =>
+        u.username.toLowerCase().indexOf(filter.toLowerCase()) >= 0 ||
```
Comment (chrisnojima) [inline shared/chat/hud/index.js]: make filter.toLowerCase() once above instead of doing it for every item

**Comment by buoyad** [comment]
Found a bug where users without `fullNames` weren't passed through to the hud. To generate the user list, we need to combine data from `fullNames` and `participants`. I used reselect to do the processing to keep it out of `mapStateToProps`. PTAL

File: shared/chat/hud/index.js [hunk 3]
```diff
@@ -87,16 +87,19 @@ type MentionHudProps = {
...
-      username: u,
-      fullName: '', // TODO
-      key: u,
+      username: u.username,
+      fullName: u.fullName,
+      key: u.username,
     }))
-    .filter(u => u.username.indexOf(filter) >= 0)
+    .filter(u => {
+      filter = filter.toLowerCase()
```
Comment (chrisnojima) [inline shared/chat/hud/index.js]: this should be hoisted to the very top of this if you're going to do it. its not strictly necessary (i'm sure this isn't a perf hot spot) but the original comment was that it gets called over and over in this map

Comment (chrisnojima) [reply shared/chat/hud/index.js]: the other option is to lowercase it in reselect or mapstatetoprops

File: shared/chat/hud/mention-hud-container.js [hunk 4]
```diff
@@ -14,11 +15,23 @@ type ConnectedMentionHudProps = {
...
+      }, [])
+    : []
+})
+
 const mapStateToProps: MapStateToProps<*, *, *> = state => {
   const inbox = getSelectedInbox(state)
-  const participants = inbox ? inbox.get('participants').toArray() : []
   return {
-    userIds: participants,
+    users: userSelector(inbox),
```
Comment (chrisnojima) [inline shared/chat/hud/mention-hud-container.js]: this is fine , you can also just chain the reselect so you pass it state and that calls another selector which pulls out inbox (more typical)

**Pull request merged**